server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -623,7 +665,35 @@ public int compare(File o1, File o2)
           if (Ints.tryParse(segmentDir.getName()) == null) {
             continue;
           }
-
+          QueryableIndex queryableIndex = null;
+          try {
+            queryableIndex = IndexIO.loadIndex(segmentDir);
+          }
+          catch (IOException e) {
+            log.error(e, "Problem loading segmentDir from disk.");
+            isCorrupted = true;
+          }
+          if (isCorrupted) {
+            try {
+              File corruptSegmentDir = computeCorruptedFileDumpDir(segmentDir, schema);
+              log.info("Renaming %s to %s", segmentDir.getAbsolutePath(), corruptSegmentDir.getAbsolutePath());
+              FileUtils.copyDirectory(segmentDir, corruptSegmentDir);
+              FileUtils.deleteDirectory(segmentDir);
+            }
+            catch (Exception e1) {
+              log.error(e1, "Failed to rename %s", segmentDir.getAbsolutePath());
+            }
+            continue;
+          }
+          BasicFileAttributes attr = Files.readAttributes(segmentDir.toPath(), BasicFileAttributes.class);
+          if (attr.creationTime().toMillis() > latestCommitTime) {
```

Comment:
Does this work on ext3 or xfs? I thought that ext4 is the only popular linux fs that stores creation time. If that's true then I don't think we should rely on it, since ext4 adoption isn't universal. I'm also a little sketched out about using the things that *are* standard on all fses, since the times in posix stat structs are all "last update" times, which could jump ahead at various times.

So, is it possible use the name of the directory, or a file in the directory, to figure out which one is newest?